### ISSUE TITLE
[Bug]: [string_match] The results are incomplete for search iterator with infix and postfix string match after flush

### ISSUE THREAD

**Issue by binbinlv** [milvus-io/milvus#30791]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20240222-16b4c9a7
- Deployment mode(standalone or cluster): both
- MQ type(rocksmq, pulsar or kafka):    all
- SDK version(e.g. pymilvus v2.0.0rc2): 2.4.0rc35
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

The results are incomplete for search iterator with infix and postfix string match after flush

```
>>> search_iterator = collection.search_iterator(vectors[:1], "float_vector", {"metric_type": "L2"}, 100, expr="varchar like '%0%'")
>>> id_array = []
>>> while True:
...     res = search_iterator.next()
...     if len(res) == 0:
...          search_iterator.close()
...          break
...     for i in range(len(res)):
...          id_array.append(res[i].id)
...
>>> len(id_array)
452
>>> collection.flush()
>>> search_iterator = collection.search_iterator(vectors[:1], "float_vector", {"metric_type": "L2"}, 100, expr="varchar like '%0%'")
>>> id_array = []
>>> while True:
...     res = search_iterator.next()
...     if len(res) == 0:
...         search_iterator.close()
...         break
...     for i in range(len(res)):
...         id_array.append(res[i].id)
...
>>> len(id_array)
83
```

### Expected Behavior

The results for search iterator with infix and postfix string match keep correct both before and after flush

### Steps To Reproduce

```python
from pymilvus import CollectionSchema, FieldSchema
from pymilvus import Collection
from pymilvus import connections
from pymilvus import DataType
from pymilvus import Partition
import numpy as np
connections.connect()

dim = 128
int64_field = FieldSchema(name="int64", dtype=DataType.INT64, is_primary=True)
float_field = FieldSchema(name="float", dtype=DataType.FLOAT)
bool_field = FieldSchema(name="bool", dtype=DataType.BOOL)
string_field = FieldSchema(name="varchar", dtype=DataType.VARCHAR, max_length=65535)
float_vector = FieldSchema(name="float_vector", dtype=DataType.FLOAT_VECTOR, dim=dim)
schema = CollectionSchema(fields=[int64_field, float_field, bool_field, string_field, float_vector])
import numpy as np
collection = Collection("test_search_iterator_collection", schema=schema)
nb = 2000
import random
default_search_params = {"metric_type": "L2", "params": {"nprobe": 100}}
vectors = [[random.random() for _ in range(dim)] for _ in range(nb)]
res = collection.insert([[i for i in range(nb)], [np.float32(i) for i in range(nb)], [np.bool_(i) for i in range(nb)], [str(i) for i in range(nb)], vectors])
index_param = {"index_type": "IVF_FLAT", "metric_type": "L2", "params": {"nlist": 100}}
collection.create_index("float_vector", index_param, index_name="a")
collection.load()
search_iterator = collection.search_iterator(vectors[:1], "float_vector", {"metric_type": "L2"}, 100, expr="varchar like '0%'")
id_array = []
while True:
    res = search_iterator.next()
    if len(res) == 0:
        search_iterator.close()
        break
    for i in range(len(res)):
        id_array.append(res[i].id)
id_array
search_iterator = collection.search_iterator(vectors[:1], "float_vector", {"metric_type": "L2"}, 100, expr="varchar like '%0'")

id_array = []
while True:
    res = search_iterator.next()
    if len(res) == 0:
        search_iterator.close()
        break
    for i in range(len(res)):
        id_array.append(res[i].id)
id_array
search_iterator = collection.search_iterator(vectors[:1], "float_vector", {"metric_type": "L2"}, 100, expr="varchar like '%0%'")
id_array = []
while True:
     res = search_iterator.next()
     if len(res) == 0:
          search_iterator.close()
          break
     for i in range(len(res)):
          id_array.append(res[i].id)
len(id_array)
collection.flush()
search_iterator = collection.search_iterator(vectors[:1], "float_vector", {"metric_type": "L2"}, 100, expr="varchar like '%0'")
id_array = []
while True:
     res = search_iterator.next()
     if len(res) == 0:
         search_iterator.close()
         break
     for i in range(len(res)):
         id_array.append(res[i].id)

len(id_array)
search_iterator = collection.search_iterator(vectors[:1], "float_vector", {"metric_type": "L2"}, 100, expr="varchar like '%0%'")
id_array = []
while True:
     res = search_iterator.next()
     if len(res) == 0:
         search_iterator.close()
         break
     for i in range(len(res)):
         id_array.append(res[i].id)
len(id_array)
```

**Comment by binbinlv** [comment]
It does not exists this issue in prefix string matching (0%)

**Comment by binbinlv** [comment]
Verified, but it is not fixed.
milvus: master-20240228-095cdbed-amd64
pymilvus: 2.4.0rc37

```
>>> search_iterator = collection.search_iterator(vectors[:1], "float_vector", {"metric_type": "L2"}, 100, expr="varchar like '%0'")
>>> id_array = []
>>> while True:
...     res = search_iterator.next()
...     if len(res) == 0:
...         search_iterator.close()
...         break
...     for i in range(len(res)):
...         id_array.append(res[i].id)
...
>>> len(id_array)
200
>>> search_iterator = collection.search_iterator(vectors[:1], "float_vector", {"metric_type": "L2"}, 100, expr="varchar like '%0%'")
>>>
>>> id_array = []
>>>
>>> while True:
...      res = search_iterator.next()
...      if len(res) == 0:
...           search_iterator.close()
...           break
...      for i in range(len(res)):
...           id_array.append(res[i].id)
...
>>>
>>> len(id_array)
452
>>> collection.flush()
>>> collection.flush()
>>> collection.flush()
>>> collection.flush()
>>> collection.flush()
>>> collection.flush()
>>> collection.flush()
>>>
>>> search_iterator = collection.search_iterator(vectors[:1], "float_vector", {"metric_type": "L2"}, 100, expr="varchar like '%0'")
>>>
>>> id_array = []
>>>
>>>
>>> while True:
...      res = search_iterator.next()
...      if len(res) == 0:
...          search_iterator.close()
...          break
...      for i in range(len(res)):
...          id_array.append(res[i].id)
...
>>> len(id_array)
33
>>> search_iterator = collection.search_iterator(vectors[:1], "float_vector", {"metric_type": "L2"}, 100, expr="varchar like '%0%'")
>>> id_array = []
>>>
>>> while True:
...      res = search_iterator.next()
...      if len(res) == 0:
...          search_iterator.close()
...          break
...      for i in range(len(res)):
...          id_array.append(res[i].id)
...
>>> len(id_array)
90
```

**Comment by longjiquan** [comment]
#30798 and #30995 is the root cause of this issue

**Comment by longjiquan** [comment]
can be closed when #30995 is fixed

**Comment by binbinlv** [comment]
OK, let this issue open as a known issue to remind user until the fix of issue #30995.

**Comment by xiaofan-luan** [comment]
From offline discussion.
Range search doesn't follow nprobe params so even if nprobe = nlist there is no ganruatee that search can get 100% result
